bonita-integration-tests/bonita-integration-tests-client/src/main/java/org/bonitasoft/engine/activity/UserTaskContractITest.java

```diff
@@ -100,8 +99,7 @@ public void shouldHandleContractInputsAtProcessLevel() throws Exception {
         final ContractDefinitionBuilder contract = builder.addContract();
         contract.addInput(numberOfDaysProcessContractData, Type.INTEGER, null);
         contract.addInput("multipleText", Type.TEXT, "a multiple text", true);
-        contract.addInput("complex", "a complex input",
-                Collections.<InputDefinition>singletonList(new InputDefinitionImpl("text", Type.TEXT, "text in complex")));
+        contract.addInput("complex", "a complex input").addChildren().addInput("text", Type.TEXT, "text in complex");
```

Comment:
What about having a addComplexInput like addFileInput returning a new builder instance. It will be more readable.